diff --git a/backend/cmd/headlamp.go b/backend/cmd/headlamp.go
index 811495890..ae21b1b85 100644
--- a/backend/cmd/headlamp.go
+++ b/backend/cmd/headlamp.go
@@ -160,6 +160,20 @@ func mustWriteFile(path string, data []byte) {
 	}
 }
 
+func makeStaticIndexWritable(staticDir string) error {
+	indexPath := filepath.Join(staticDir, "index.html")
+	data, err := os.ReadFile(indexPath)
+	if err != nil {
+		return err
+	}
+	if err = os.Remove(indexPath); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+	}
+	return os.WriteFile(indexPath, data, fs.FileMode(0o600))
+}
+
 func makeBaseURLReplacements(data []byte, baseURL string) []byte {
 	replaceURL := baseURL
 	if baseURL == "" {
@@ -1158,6 +1172,12 @@ func startServer(config *HeadlampConfig) {
 			return
 		}
 
+		err = makeStaticIndexWritable(dir)
+		if err != nil {
+			logger.Log(logger.LevelError, nil, err, "Failed to make copied static index writable")
+			return
+		}
+
 		config.StaticDir = dir
 	}
 
